### ISSUE TITLE
pre: mod_chmod

### ISSUE THREAD

**Issue by sirmarksalot** [glftpd/foo-tools#6]
How is ```mod_chmod.so``` suppose to work?

I always get failed to chmod error messages.
```chmod file.txt failed!```

https://github.com/glftpd/foo-tools/blob/73e6265afcc797232adfc0ce89cb9f52231e6d51/src/pre/foo-pre.c#L1084
https://github.com/glftpd/foo-tools/blob/73e6265afcc797232adfc0ce89cb9f52231e6d51/src/pre/foo-pre.c#L1138
the modules are called as a non superuser. chmod is unable to change the permissions of directories that are owned by uid 0

**Comment by silv3rr** [comment]
Try https://github.com/glftpd/foo-tools/blob/73e6265afcc797232adfc0ce89cb9f52231e6d51/doc/README.foo-pre#L132

**Comment by sirmarksalot** [comment]
@silv3rr yes that is already correctly set and works fine. chown works without issue.

mod_chmod works aswell after foo-pre.c is edited and "setuid(olduid);" is moved from line 1084 to 1139
